napari/utils/interactions.py

```diff
@@ -309,6 +309,28 @@ def __init__(self, shortcut: KeyBindingLike) -> None:
         if error:
             warnings.warn(error_msg, UserWarning, stacklevel=2)
 
+    @staticmethod
+    def parse_platform(text: str) -> str:
+        """
+        Parse a current_platform_specific shortcut, and return a canonical
+        version separated with dashes.
+
+        This replace platform specific symbols, like ↵ by Enter,  ⌘ by Command on MacOS....
+        """
+        # edge case, shortcut combinaison where `+` is a key.
+        # this should be rare as on english keyboard + is Shift-Minus.
+        # but not unheard of. In those case `+` is always at the end with `++`
+        # as you can't get two non-modifier keys,  or alone.
+        if text == '+':
+            return text
+        if joinchar == "+":
+            text.replace('++', '+Plus')
+            text.replace('+', '')
+            text.replace('Plus', '+')
+        for k, v in KEY_SYMBOLS.items():
+            text = text.replace(v, k + '-')
```

Comment:
@psobolewskiPhD I can reproduce your issue.
I'm not sure this is the _right_ fix, but it's one way to fix your issue. Basically this was trying to bind `"Space-"` which is invalid, so here I only append the extra hyphen if the keybind is longer than the symbol.

```suggestion
        for k, v in KEY_SYMBOLS.items():
            if len(text) > len(v):
                text = text.replace(v, k + '-')
            else:
                text = text.replace(v, k)
```

Reply:
I went for a different fix, because in this case of `⌘⌥` that would not have word, and returned `Meta-Alt-` which also would be invalid.

Reply:
Ok, that explains why it's things like space, backspace, etc. that trigger the bug!
I'm not sure what the hyphen is for at all—I never see it—and with this PR I never see the `+` joining keys either, like e.g. Command+M on main.

Previous to the app-model keybindings it looked like the settings.yaml had `-` as the join. 
Currently, I see `+` in my settings.yaml
Still, multi key bindings work with this PR, e.g. `Shift+Meta+Z` (the `+` don't show while editing, which seems like a macOS specific setting.)

I think the `-` might be a leftover and maybe just needs to be removed.

Reply:
Ah, good point - I just saw your change and it looks good to me!

Reply:
`-` is at least still present in  test , and it is still likely used in extensions I guess. 

I'm playing around with NewType to see if we can avoid using 'Str' everywhere. 

https://github.com/Carreau/napari/pull/new/ntt